src/api/firebase.js

```diff
@@ -170,25 +171,23 @@ export async function shareList(listPath, currentUserId, recipientEmail) {
  */
 export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	const listCollectionRef = collection(db, listPath, 'items');
-	// TODO: Replace this call to console.log with the appropriate
-	// Firebase function, so this information is sent to your database!
 	return addDoc(listCollectionRef, {
 		dateCreated: new Date(),
-		// NOTE: This is null because the item has just been created.
-		// We'll use updateItem to put a Date here when the item is purchased!
 		dateLastPurchased: null,
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listPath, itemId) {
+	const itemDocRef = doc(db, listPath, 'items', itemId);
+	const datePurchased = new Date();
+	await updateDoc(itemDocRef, {
+		dateLastPurchased: datePurchased,
+		totalPurchases: increment(1),
+		dateNextPurchased: getFutureDate(1),
+	});
 }
```

Comment:
Nice work implementing the correct Firestore functions! 🙌🏽 I do have one concern: the `updateItem` function seems a bit rigid when it might be intended to handle a variety of updates to the item. To enhance its flexibility and reusability, I suggest passing the `updates` as a prop. This way, we can easily adapt the function to different update scenarios! 🚀